### PR TITLE
Tutorial store tests

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier

For #113

Describe your changes:
Adds tests for the TutorialStore

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by srallen** [comment]
Rebased to get the fixed LCV tests and travis configs.

File: packages/lib-classifier/src/store/TutorialStore.js [hunk 1]
```diff
@@ -76,10 +76,9 @@ const TutorialStore = types
         try {
           const response = yield tutorials.getAttachedImages({ id: tutorial.id })
           const { media } = response.body
-          self.setMediaResources(media)
+          if (media && media.length > 0) self.setMediaResources(media)
         } catch (error) {
           console.error(error)
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/TutorialStore.js]: Why is the error state removed here? Are you ignoring 500 errors etc. and just displaying the tutorial without attached media if Panoptes is broken/unavailable?

File: packages/lib-classifier/src/store/TutorialStore.spec.js [hunk 2]
```diff
@@ -1,74 +1,337 @@
...
+    expect(rootStore.tutorials.loadingState).to.equal(asyncStates.initialized)
+  })
+
+  it('should set the tutorial if there is a workflow', function (done) {
+    rootStore = RootStore.create({
+      tutorials: TutorialStore.create(),
+      workflows: WorkflowStore.create()
+    }, { authClient: authClientStubWithUser, client: clientStub })
+    rootStore.workflows.setActive(workflow.id)
+      .then(() => {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/TutorialStore.spec.js]: Minor point: Arrow functions are discouraged in mocha tests, but my understanding is that's only a big deal for tests that use `this`.

File: packages/lib-classifier/src/store/TutorialStore.spec.js [hunk 3]
```diff
@@ -1,74 +1,337 @@
...
+      })
+
+      rootStore.workflows.setActive(workflow.id).then(() => {
+        expect(rootStore.tutorials.loadingState).to.equal(asyncStates.error)
+      }).then(done, done)
     })
+  })
+
+  describe('Actions > fetchMedia', function () {
+    it('should not call setMediaResources if there is not media in the response', function (done) {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/TutorialStore.spec.js]: Typo: 'no media in the response'?

File: packages/lib-classifier/src/store/TutorialStore.spec.js (hunk 2 above)
Comment (srallen) [reply packages/lib-classifier/src/store/TutorialStore.spec.js]: I think it's an issue between expectations or describe blocks if you use arrow functions, but I think it's fine within a single expectation. The `it` function that is running this promise here is not using it.

File: packages/lib-classifier/src/store/TutorialStore.js (hunk 1 above)
Comment (srallen) [reply packages/lib-classifier/src/store/TutorialStore.js]: Yep, we don't care if the image request fails. This is the current behavior in PFE. The `catch` there just returns an empty array and doesn't even log the error at all. I think if at some point we setup an error logging service, we'll want to add that here, but we don't want to stop the tutorial from rendering.

**Pull request merged**